src/packages/emmett-mongodb/src/eventStore/mongoDBEventStore.ts

```diff
@@ -378,6 +408,122 @@ class MongoDBEventStoreImplementation implements MongoDBEventStore {
     if (!this.isClosed) await this.client.connect();
     return this.client;
   };
+
+  private async findOneInlineProjection<
+    Doc extends Document,
+    Projection = MongoDBReadModel<Doc>,
+  >(
+    streamType: StreamType,
```

Comment:
**SUGGESTION:** I think that we could also allow passing stream name to find one, then we'd automatically allow selecting the read model by the ID without additional parameters. So we could do something like:

````ts
      await eventStore.projections.inline.findOne<ShoppingCartShortInfo>(
        streamName
      );
```

Thoughts?

Or we could introduce the conventional ID for projection name like we did for stream name. It could be streamType: streamId: projectionName. Then, we could allow passing stream type, stream name, or projection name in this first parameter. Thoughts?

Reply:
Yeah I like that. Right now I left that as an available option you could provide in the query itself, but I think for these stream level properties, maybe we make the first param an object that is either the `streamType` or can be `streamName` or optional `streamId`

Reply:
```typescript
// querying over whole collection
projections.inline.findOne({ streamType }, ...)

// finding specific stream projection
projections.inline.findOne({ streamName }, ...)

// streamId also optional, maybe just for convenience whether you already
// have a formatted name or just the parts.
projections.inline.findOne({ streamType, streamId }, ...)
```

Reply:
Yes, an object also works; we can always make it either an object or a conventional ID; depending on someone's preferences, they can use it either way. The object will probably be the most accessible to folks.